vendor/parser-extended/parser.mly

```diff
@@ -2266,10 +2269,8 @@ expr:
         Pexp_letopen(od, $7), $4 }
   | FUNCTION ext_attributes match_cases
       { Pexp_function $3, $2 }
-  | FUN ext_attributes fun_param fun_def
+  | FUN ext_attributes expr_fun_param fun_def
       { Pexp_fun($3, $4), $2 }
-  | FUN ext_attributes LPAREN TYPE lident_list RPAREN fun_def
-      { (mk_newtypes ~loc:$sloc $5 $7).pexp_desc, $2 }
```

Comment:
This diff might be hard to maintain, I think I'd prefer some code duplication instead.

Reply:
Theses cases are already merged upstream, there are new rules like `fun_param_as_list` and `fun_params`. I will try to get closer to these parser rules.